### PR TITLE
risc-v/mpfs: apply IHC review fixes

### PR DESCRIPTION
**Pull request by eenurkka**
PR#6249 was already merged without the review fixes. Provide the
fixes here on a separate patch.

Co-authored-by: Petro Karashchenko <petro.karashchenko@gmail.com>
Signed-off-by: Eero Nurkkala <eero.nurkkala@offcode.fi>

## Summary

Apply the review fixes here, better late than not at all to keep the code in line

## Impact

No impact

## Testing

Tested on Polarfire Icicle

### CODE REVIEW

File: arch/risc-v/src/mpfs/mpfs_ihc.c [hunk 1]
```diff
@@ -1248,19 +1248,18 @@ int mpfs_ihc_init(void)
...
 }
 
 /****************************************************************************
  * Name: up_addrenv_pa_to_va
  *
  * Description:
  *   This is needed by openamp/libmetal/lib/system/nuttx/io.c. The
- *   physical memory is mapped as virtual.
+ *   physical memory is mapped as virtual.  FAR is here to match the original
+ *   declaration in arch.h.
```
Comment (pkarashchenko) [inline arch/risc-v/src/mpfs/mpfs_ihc.c:1262]:
```suggestion
 *   physical memory is mapped as virtual.
```

File: arch/risc-v/src/mpfs/mpfs_ihc.c [hunk 2]
```diff
@@ -1272,5 +1271,5 @@ uintptr_t up_addrenv_va_to_pa(FAR void *va_)
 
 FAR void *up_addrenv_pa_to_va(uintptr_t pa)
```
Comment (pkarashchenko) [inline arch/risc-v/src/mpfs/mpfs_ihc.c:1272]:
```suggestion
void *up_addrenv_pa_to_va(uintptr_t pa)
```

File: arch/risc-v/src/mpfs/mpfs_ihc.c (hunk 1 above)
Comment (eenurkka) [reply arch/risc-v/src/mpfs/mpfs_ihc.c]: Changed, thanks

File: arch/risc-v/src/mpfs/mpfs_ihc.c [hunk 3]
```diff
@@ -1248,11 +1248,9 @@ int mpfs_ihc_init(void)
  *
  ****************************************************************************/
 
-uintptr_t up_addrenv_va_to_pa(FAR void *va_)
+uintptr_t up_addrenv_va_to_pa(FAR void *va)
```
Comment (pkarashchenko) [inline arch/risc-v/src/mpfs/mpfs_ihc.c:1251]:
```suggestion
uintptr_t up_addrenv_va_to_pa(void *va)
```

Comment (eenurkka) [reply arch/risc-v/src/mpfs/mpfs_ihc.c]: Fixed that as well, now absolutely no FARs.

**Review by pkarashchenko — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

File: arch/risc-v/src/mpfs/mpfs_ihc.c [hunk 4]
```diff
@@ -163,19 +163,19 @@ static int mpfs_rptun_register_callback(struct rptun_dev_s *dev,
...
+static struct mpfs_queue_table      g_mpfs_virtqueue_table[VRINGS];
+static struct mpfs_rptun_shmem_s    g_shmem;
+static struct rpmsg_device         *g_mpfs_rpmsg_device;
+static struct rpmsg_virtio_device  *g_mpfs_virtio_device;
 
-static sem_t   g_mpfs_ack_lock      = SEM_INITIALIZER(1);
-static sem_t   g_mpfs_rx_lock       = SEM_INITIALIZER(1);
-static struct  list_node g_dev_list = LIST_INITIAL_VALUE(g_dev_list);
+static sem_t  g_mpfs_ack_lock      = SEM_INITIALIZER(1);
+static sem_t  g_mpfs_rx_sig        = SEM_INITIALIZER(1);
```
Comment (pkarashchenko) [inline arch/risc-v/src/mpfs/mpfs_ihc.c:173]: @eenurkka are you sure that `g_mpfs_rx_sig` should be inited to `SEM_INITIALIZER(1)` and not to `SEM_INITIALIZER(0)`?

Comment (pkarashchenko) [reply arch/risc-v/src/mpfs/mpfs_ihc.c:173]: Same for `g_mpfs_ack_lock` that also seems to be used for signalling (maybe can be renamed to `g_mpfs_ack_sig`)

Comment (eenurkka) [reply arch/risc-v/src/mpfs/mpfs_ihc.c:173]: Could be, I grepped *.c files and figured everybody else uses SEM_INITIALIZER(1), so I went with that. Will check.

Comment (pkarashchenko) [reply arch/risc-v/src/mpfs/mpfs_ihc.c:173]:
```
static struct ramlog_dev_s g_sysdev =
{
#ifndef CONFIG_RAMLOG_NONBLOCKING
  0,                             /* rl_nwaiters */
#endif
  CONFIG_RAMLOG_BUFSIZE,         /* rl_head */
  CONFIG_RAMLOG_BUFSIZE,         /* rl_tail */
  SEM_INITIALIZER(1),            /* rl_exclsem */
#ifndef CONFIG_RAMLOG_NONBLOCKING
  SEM_INITIALIZER(0),            /* rl_waitsem */
#endif
  CONFIG_RAMLOG_BUFSIZE,         /* rl_bufsize */
  g_sysbuffer                    /* rl_buffer */
};
#endif
```
so `rl_exclsem` is mutual exclusion access guard while `rl_waitsem` is a signalling semaphore

Comment (eenurkka) [reply arch/risc-v/src/mpfs/mpfs_ihc.c:173]: Yes, it would look like the zero is better here

Comment (eenurkka) [reply arch/risc-v/src/mpfs/mpfs_ihc.c:173]: Created a separate PR#6261

**Pull request merged**